### ISSUE TITLE
Convergence criteria

### ISSUE THREAD

**Issue by jgallowa07** [matsengrp/multidms#137]
This issue is to note that `jaxopt` the package we use for optimizing our model, is merging into `optax`. While this poses no problem for the software as it stands it would certainly be desirable to make the switch once they merge in `ProximalGradient`.

This would solve current problems with the way [we do training steps](https://github.com/matsengrp/multidms/blob/58891661775fb1e50c39b9947775d70d508c0452/multidms/model_collection.py#L192) just to get a convergence line. This leads to non-deterministic results (i.e. 1000 iterations for one step != 100 iterations for 10 steps). and future development, more generally.

**Comment by jgallowa07** [comment]
> I wouldn't call this nondeterministic, but it is a deterministic function of how we partition iterations into calls to run

This is great - yes that makes sense to me. luckily we're [actually deterministic](https://github.com/matsengrp/multidms/blob/e376173d3aaf99c9a7644ce208db0061c5d6306c/tests/test_data.py#L349) otherwise. 

> We could define our own run command that iterates calls to update until state.error = tol or state.iteration = maxiter, and outputs loss trajectory data from each iterate.

This is amazing. It'll be until after re submission that I'll be able to do this - would happily assist if you had the bandwidth to PR.

**Comment by jgallowa07** [comment]
I'll just note that this update will probably help with some issues I seem to be running into with the spike analysis. 

I suspect this has to do with the tolerance, learning rate, memory and other things that may easier to control with a custom `update()` loop - but It seems the model fits are not so robust after many training iterations. For context, we run the spike models for 30 independent rounds of 1000 iterations each (30K, total). These have the default tolerance set to 1e-4. 

![Screenshot from 2024-03-05 06-44-15](https://github.com/matsengrp/multidms/assets/23087427/99ca6421-0786-4f16-a0e4-7a03a3017abf)

However, when we run those same models a single round of 100K iterations (everything else the same), 

![Screenshot from 2024-03-05 06-57-43](https://github.com/matsengrp/multidms/assets/23087427/4da12e20-4d99-4f3b-996b-f3d95e4e452d)

we can see some of the models have certainly over-fit to their data. I wonder how exactly the tolerance works with penalties? could it be the case that penalties added to the total cost is effecting the potential for the model to quit early?  

This could also be related to #133 - as a ridge does seem to again, stabilize things. More testing will be needed here.

**Comment by jgallowa07** [comment]
Relevant to this issue, it should be noted that the primary difference between single, and multi-step models optimization is the FISTA acceleration. In the single step models, the learning rate is reset at each step. When acceleration is turned off, these two approaches yield identical results.
